component/src/test/java/org/cbioportal/genome_nexus/util/GenomicVariantUtilTest.java

```diff
@@ -186,4 +186,46 @@ public void testMafToGenomicVariant() {
         assertEquals("G", two.getRef());
         assertEquals("T", two.getAlt());
     }
+
+    @Test
+    public void testProvidedReferenceAlleleFromHgvs() {
+        // The tested function only extracts specified reference alleles. Position ranges are not examined for correct count.
+        String NONE = "";
+        LinkedHashMap<String, String> testCases = new LinkedHashMap<String, String>();
+        // Substitution cases
+        testCases.put("1g.1A>G", "A");
+        testCases.put("1g.1T>G", "T");
+        testCases.put("1g.1C>G", "C");
+        testCases.put("1g.1G>G", "G");
+        testCases.put("1g.1G>G ", "G"); // trailing whitespace is acceptable
+        testCases.put("1g.1C>", "C"); // reference allele was specified, even if required tumor seq allele was missing
+        testCases.put("1g.1AA>G", NONE); // too many nucleotides in reference allele
+        testCases.put("1g.1>G", NONE); // reference allele missing
+        testCases.put("1g.1>", NONE); // both alleles missing
+        // Deletion-Insertion cases
+        testCases.put("1g.1delinsT", NONE);
+        testCases.put("1g.1delinsT ", NONE);
+        testCases.put("1g.1delAinsT", "A");
+        testCases.put("1g.1delAinsT ", "A");
+        testCases.put("1g.1delAins", "A"); // tumor seq allele
+        testCases.put("1g.1insTdelA", "A"); // although order of ins-del is invalid, the specified reference allele is returned
+        testCases.put("1g.1_2delAAinsTT", "AA");
+        testCases.put("1g.1_3delAAAinsT", "AAA");
+        testCases.put("1g.1_2delinsTT", NONE);
+        testCases.put("1g.1_2delAinsTT", "A"); // although nucleotide count does not match reference allele size, the tested function will extract the specified Reference Allele
+        testCases.put("1g.1delAAinsTT", "AA"); // although nucleotide count does not match reference allele size, the tested function will extract the specified Reference Allele
```

Comment:
these are nice but as mentioned above the reference allele in new formats of hgvs is removed for delins. Nice to have tho